src/Magento/FunctionalTestingFramework/StaticCheck/TestDependencyCheck.php

```diff
@@ -0,0 +1,385 @@
+     * Transactional Array to keep track of what dependencies have already been extracted.
+     * @var array
+     */
+    private $alreadyExtractedDependencies;
+
+    /**
+     * Checks test dependencies, determined by references in tests versus the dependencies listed in the Magento module
+     *
+     * @param InputInterface $input
+     * @return string
+     * @SuppressWarnings(PHPMD.CyclomaticComplexity)
+     */
+    public function execute(InputInterface $input)
+    {
+        MftfApplicationConfig::create(
+            true,
+            MftfApplicationConfig::UNIT_TEST_PHASE,
+            false,
+            false
+        );
+
+        ModuleResolver::getInstance()->getModulesPath();
+        if (!class_exists('\Magento\Framework\Component\ComponentRegistrar')) {
+            return "TEST DEPENDENCY CHECK ABORTED: MFTF must be attached or pointing to Magento codebase.";
+        }
+        $registrar = new \Magento\Framework\Component\ComponentRegistrar();
+        $this->moduleNameToPath = $registrar->getPaths(\Magento\Framework\Component\ComponentRegistrar::MODULE);
+        $this->moduleNameToComposerName = $this->buildModuleNameToComposerName($this->moduleNameToPath);
+        $this->flattenedDependencies = $this->buildComposerDependencyList();
+
+        $allModules = ModuleResolver::getInstance()->getModulesPath();
+        $filePaths = [
+            DIRECTORY_SEPARATOR . 'Test' . DIRECTORY_SEPARATOR,
+            DIRECTORY_SEPARATOR . 'ActionGroup' . DIRECTORY_SEPARATOR,
+            DIRECTORY_SEPARATOR . 'Data' . DIRECTORY_SEPARATOR,
+        ];
+        // These files can contain references to other modules.
+        $testXmlFiles = $this->buildFileList($allModules, $filePaths[0]);
+        $actionGroupXmlFiles = $this->buildFileList($allModules, $filePaths[1]);
+        $dataXmlFiles= $this->buildFileList($allModules, $filePaths[2]);
+
+        $testErrors = [];
+        $testErrors += $this->findErrorsInFileSet($testXmlFiles);
+        $testErrors += $this->findErrorsInFileSet($actionGroupXmlFiles);
+        $testErrors += $this->findErrorsInFileSet($dataXmlFiles);
+
+        //print all errors to file
+        return $this->printErrorsToFile($testErrors);
+    }
+
+    /**
+     * Finds all reference errors in given set of files
+     * @param Finder $files
+     * @return array
+     * @throws \Magento\FunctionalTestingFramework\Exceptions\TestReferenceException
+     * @throws \Magento\FunctionalTestingFramework\Exceptions\XmlException
+     */
+    private function findErrorsInFileSet($files)
+    {
+        $testErrors = [];
+        foreach ($files as $filePath) {
+            $modulePath = dirname(dirname(dirname(dirname($filePath))));
+            $moduleFullName = array_search($modulePath, $this->moduleNameToPath) ?? null;
+            // Not a module, is either dev/tests/acceptance or loose folder with test materials
+            if ($moduleFullName == null) {
+                continue;
+            }
```

Comment:
I am running standalone mftf, and I have a test module under mftf's dev/tests/functional/tests/MFTF/CatalogTest which has been scanned but it should not.

Reply:
Will check/resolve this.